src/server/models/admin.js

```diff
@@ -1 +1,62 @@
-module.exports = {};
+const {
+  collection,
+  getDoc,
+  getDocs,
+  doc,
+  setDoc,
+  updateDoc,
+  serverTimestamp,
+} = require('firebase/firestore');
+const dbPromise = require('../../config/firebase');
+
+module.exports = {
+  get: async () => {
+    try {
+      const db = await dbPromise;
+      const superUserRef = collection(db, 'superUsers');
+      const superUserSnapShot = await getDocs(superUserRef);
+      const superUsers = superUserSnapShot.docs.map(
+        (d) => ({
+          id: d.id,
+          ...d.data(),
+        }),
+      );
+
+      return superUsers;
+    } catch (error) {
+      throw new Error(error);
+    }
+  },
+  post: async (payload) => {
+    try {
+      const db = await dbPromise;
+      const superUsersRef = collection(db, 'superUsers');
+
+      return setDoc(doc(superUsersRef), {
+        ...payload,
+        createdAt: serverTimestamp(),
+        updatedAt: serverTimestamp(),
```

Comment:
Very cool you added the timestamp using Firestore's serverTimestamp(). I'll look into that.